### PR TITLE
cli: Skip showing rules and products for pattern mode

### PR DESCRIPTION
**Pull request by zzeleznick**
## Description
Per user feedback, we should limit the displayed output for `-e/--pattern` invocations.

### Example

`$ semgrep -l ocaml --pattern 'env_or'`

**Before:** Includes product enablement status under the Semgrep CLI Header
```                            
✔ Semgrep OSS
  ✔ Basic security coverage for first-party code vulnerabilities.
                                              
✘ Semgrep Code (SAST)
  ✘ Find and fix vulnerabilities in the code you write with advanced scanning and expert security rules.
                                                     
✘ Semgrep Supply Chain (SCA)
  ✘ Find and fix the reachable vulnerabilities in your OSS dependencies.
```
**After:** Remove the Semgrep CLI Header and progress bar

### CODE REVIEW

File: cli/bin/sg [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python3
```
Comment (aryx) [inline cli/bin/sg:1]: This requires a changelog entry telling people semgrep is now also accessible as the shorter
'sg'
Maybe @milanwilliams has an opinion about this (I am personally in favor).

**Review by aryx — APPROVED** [review]

File: cli/tests/e2e/snapshots/test_dependency_aware_rules/test_dependency_aware_rules/rulesdependency_awarego-sca.yaml-dependency_awarego_toolchain/results.txt [hunk 2]
```diff
@@ -50,16 +50,17 @@ SEMGREP_USER_AGENT_APPEND="pytest" SEMGREP_SETTINGS_FILE="<MASKED>" SEMGREP_VERS
...
-          "reachable": false,
+          "reachable": true,
           "sca_finding_schema": 20220913
         },
-        "severity": "WARNING"
+        "severity": "WARNING",
+        "validation_state": "NO_VALIDATOR"
       },
-      "path": "targets/dependency_aware/go_toolchain/go.mod",
+      "path": "targets/dependency_aware/go_toolchain/sca.go",
```
Comment (aryx) [inline cli/tests/e2e/snapshots/test_dependency_aware_rules/test_dependency_aware_rules/rulesdependency_awarego-sca.yaml-dependency_awarego_toolchain/results.txt]: This seems unrelated. Maybe merge develop and you should not need this?

Comment (zzeleznick) [reply cli/tests/e2e/snapshots/test_dependency_aware_rules/test_dependency_aware_rules/rulesdependency_awarego-sca.yaml-dependency_awarego_toolchain/results.txt]: Indeed. @mjambon fixed this test snapshot later in `Update test snapshot (how did it get in?)` #9119. 

I also asked about this in our [#team-supply-chain](https://semgrepinc.slack.com/archives/C032JB2HRDK/p16984501901239590) channel.

File: cli/bin/sg (hunk 1 above)
Comment (zzeleznick) [reply cli/bin/sg:1]: I'm open to adding a changelog entry in this PR, but I figured we could defer before we make an official announcement (e.g. as noted in the public-facing docs).

Earlier I had set an alias via my `.zshrc` config
```sh
alias sg="/opt/homebrew/bin/semgrep"
```
and this new file is functionally equivalent (w/ one fewer step).
